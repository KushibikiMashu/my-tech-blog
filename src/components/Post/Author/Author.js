import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import { getContactHref } from '../../../utils';
import styles from './Author.module.scss';

export const PureAuthor = ({ data }) => {
  const { author } = data.site.siteMetadata;

  return (
    <div className={styles['author']}>
      <p className={styles['author__bio']}>
        {author.bio}

        <div className={styles['author__bio-twitter']}>
          Twitter:{' '}
          <a
            href={getContactHref('twitter', author.contacts.twitter)}
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>@Panda_Program</strong>
          </a>
        </div>
      </p>
    </div>
  );
};

export const Author = (props) => (
  <StaticQuery
    query={graphql`
      query AuthorQuery {
        site {
          siteMetadata {
            author {
              name
              bio
              contacts {
                twitter
              }
            }
          }
        }
      }
    `}
    render={(data) => <PureAuthor {...props} data={data}/>}
  />
);

export default Author;
