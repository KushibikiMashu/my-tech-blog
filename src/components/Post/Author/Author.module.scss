@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.author {
  border-top: 1px solid $color-gray-border;
  max-width: $layout-post-width;
  padding-left: 20px;
  display: flex;
  padding-top: 20px;
  @include line-height(1);
  @include margin-top(1);
  @include margin-bottom(3);

  &__photo {
    display: flex;
    align-items: center;

    &-image {
      margin-right: 0;
      margin-bottom: 0;
      min-width: 75px;
    }
  }

  &__bio {
    @include margin-top(1);
    padding-left: 20px;

    &-twitter {
      @include margin-bottom(0.5);
      display: block;
      text-decoration: none;
    }
  }
}

@include breakpoint-sm {
  .author {
    margin-left: auto;
    margin-right: auto;
    @include margin-bottom(2);
  }
}