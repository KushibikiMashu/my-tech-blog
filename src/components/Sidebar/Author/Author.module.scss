@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.author {
  &__photo {
    display: inline-block;
    margin-bottom: 0;
    background-clip: padding-box
  }

  &__title {
    font-size: $typographic-base-font-size * 1.125;
    font-weight: 600;
    @include line-height(1.125);
    @include margin(0.5, 0, 0, 0);

    &-link {
      color: $color-base;

      &:hover,
      &:focus {
        color: $color-base
      }

    }

  }

  &__subtitle {
    color: $color-base;
    font-size: $typographic-base-font-size * 0.9;
    @include line-height(0.8);
    @include margin-bottom(1.25)
  }
}
