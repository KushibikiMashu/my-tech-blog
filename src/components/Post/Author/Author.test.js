// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Author from './Author';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('Author', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const props = {
      name: 'string',
      bio: 'string',
      photo: 'string',
      contacts: {
        twitter: 'string',
        github: 'string',
      }
    };

    const tree = renderer.create(<Author {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
