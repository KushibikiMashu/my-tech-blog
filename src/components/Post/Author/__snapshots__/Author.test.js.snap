// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div>
  <div
    className="author"
  >
    <div
      className="author__photo"
    >
      <img
        alt="Test name"
        className="author__photo-image"
        height="75"
        src="/test.jpg"
        width="75"
      />
    </div>
    <p
      className="author__bio"
    >
      <strong>
        プログラミングをするパンダ
      </strong>
      <br />
      Test bio
      <div
        className="author__bio-twitter"
      >
        Twitter:
         
        <a
          href="https://www.twitter.com/#"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            @Panda_Program
          </strong>
        </a>
      </div>
    </p>
  </div>
</div>
`;
